features/shell/src/main/java/org/opennms/alec/features/shell/ProcessAlarms.java

```diff
@@ -98,19 +100,34 @@ public class ProcessAlarms implements Action {
     @Option(name = "--beta", description = "Define beta")
     private double beta = 0.55257784d;
 
+    @Option(name = "--token", description = "Azure token for remote model")
+    private String token;
+
+    @Option(name = "--uri", description = "Azure uri for remote model")
+    private String uri;
+
     @Override
     public Object execute() throws Exception {
         EngineFactory engineFactory = getEngineFactory();
-        final DistanceMeasureFactory distanceMeasureFactory = getDistanceMeasureFactory();
+
         if ("dbscan".equals(engineFactory.getName())) {
+            final DistanceMeasureFactory distanceMeasureFactory = getDistanceMeasureFactory();
             ((DBScanEngineFactory) engineFactory).setDistanceMeasureFactoryName(distanceMeasureFactory.getName());
             ((DBScanEngineFactory) engineFactory).setEpsilon(epsilon);
             ((DBScanEngineFactory) engineFactory).setAlpha(alpha);
             ((DBScanEngineFactory) engineFactory).setBeta(beta);
+        } else if ("deeplearning".equals(engineFactory.getName()) && !token.isEmpty() && !uri.isEmpty()) {
+            ((DeepLearningEngineFactory) engineFactory).setToken(token);
+            ((DeepLearningEngineFactory) engineFactory).setUri(uri);
+            ((DeepLearningEngineFactory) engineFactory).setRemote(true);
+        } else if ("deeplearning".equals(engineFactory.getName())) {
+            ((DeepLearningEngineFactory) engineFactory).setToken("");
+            ((DeepLearningEngineFactory) engineFactory).setUri("");
+            ((DeepLearningEngineFactory) engineFactory).setRemote(false);
         }
```

Comment:
This feels a little like a code smell, not sure if there's a way to hide the "engine-specific" things into the engine/enginefactory instead of making a potentially ever-growing if/else if chunk as you add more engines...

That said it's not a deal-breaker.